tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+# Contributing
+
+This page describes how to contribute to [APIView](../../../src//dotnet/APIView/APIViewWeb/CONTRIBUTING.md) language level parsers.
+Specifically how to create or update a language parser to produce tree style tokens for APIView.
+
+## Creating Tree Style Tokens
+The main idea is to capture the hierachy of the API using a tree data structure, then maintain a flat list of tokens for each node of the tree.
+
+![APITree](APITree.svg)
+
+Each tree node has top tokens which should be used to capture the main tokens on the node, these can span multiple lines. If the language requires it use the bottom tokens to capture tokens that closes out the node, this is usually just the closing bracket and/or empty lines.
+
+- Here are the models needed
+  ```
+  object APITreeNode
+    string Name
+    string Id
+    string Kind
+    Set<string> Tags
+    Dictionary<string, string> Properties
+    List<StructuredToken> TopTokens
+    List<StructuredToken> BottomTokens
+    List<APITreeNode> Children
+
+  object StructuredToken
+    string Value
+    string Id
+    StructuredTokenKind Kind
+    Set<string> Tags
+    Dictionary<string, string> Properties 
+    Set<string> RenderClasses 
+
+  enum StructuredTokenKind
+    Content
+    LineBreak
+    NoneBreakingSpace
+    TabSpace
+    ParameterSeparator
+    Url
+  ```
+### APITreeNode
+Each module of the API (namespace, class, method) should be its own node. Members of a module (methods, in a class), (classes in a namespace) should be added as child nodes of its parent module.
+
+Sort each node at each level of the tree by your desired property, this is to ensure that difference in node order does not result in diff.
+
+Ensure each node has an Id. The combination of `Id`, `Kind` and `SubKind` should make the node unique across all nodes in the tree. This is very important. For example a class and a method can potentally have the same Id, but the kind should diffrentiate them from each other.
+
+- `Name (n)`  : The name of the tree node which will be used as label for the API Navigation. Generally use the name of the module (class, method)
+- `Id (i)` : Id of the node, which should be unique at the node level. i.e. unique among its siblings. Use whatever existing parser is assigning to DefinitionId for the main Token of the node. Each node must have an Id.
+- `Kind (k)` : What kind of node is it. Please ensure you set a Kind for each node. Using any of the following `assembly`, `class`, `delegate`, `enum`, `interface`, `method` , `namespace`, `package`, `struct`, `type` will get you the corresponding default icons for the page navigation but feel free to use something language specific.
```

Comment:
There should just be an enum somewhere that says what the supported NodeKinds are. What good is it to use something "language-specific" if it means I get no icon and APIView has no idea what to do with it?

Reply:
If you use something specific, you can let us know and we will support it. If you don't want to do that just stick to the list documented.